### ISSUE TITLE
Unsatisfiable constraints in Tabs

### ISSUE THREAD

**Issue by mrocklin** [bokeh/bokeh#6553]
I'm trying to use Panels and Tabs, similar to [this example](http://bokeh.pydata.org/en/latest/docs/user_guide/interaction/widgets.html#tab-panes).  

My code looks like the following:

    from bokeh.models.widgets import Panel, Tabs
    panels = [Panel(child=figs[0], title='log scale'), 
              Panel(child=figs[1], title='linear scale')]
    tabs = Tabs(tabs=panels)

When I show either of my figures (in `figs`) they look great.  However when I show the tabs object I get an "unsatisfiable constraint" error on the client side.  Is this a known issue?

![image](https://user-images.githubusercontent.com/306380/27608285-e27d7000-5b54-11e7-8458-5c7bc826a9ee.png)

I'm on 0.12.6.  I apologize that I don't have a minimal example.

**Comment by mattpap** [comment]
I haven't seen this error in tabs context recently, but as tabs are more broken than working, it's not that unexpected. Though this particular code should work. There must must be something more going on. How do you configure your figures? Do you set `min_border`, `toolbar_sticky`, etc.?

**Comment by mrocklin** [comment]
I apologize for the code dump:

```python
for axis_type in ['log', 'linear']:
    fig = figure(title='Tasks: Embarrassingly parallel', sizing_mode='scale_width', 
                 x_axis_type=axis_type, y_axis_type=axis_type, height=500, width=500)
    for i, name in enumerate(names):
        x = df3.loc['tasks', name].n.values
        y = df3.loc['tasks', name].rate.values
        fig.line(x=x, y=y, color=colors[i], legend=legends[i])
        fig.circle(x=x, y=y)

    for name in names:
        # Add in perfect scaling line
        x = df3.loc['tasks', name].n.values
        y = df3.loc['tasks', name].rate.values
        mn = x.min()
        mx = x.max()
        slope = y[0] / mn
        fig.line(x=[0, mx], y=[0, slope * mx], color='gray', line_dash='dashed')

    fig.y_range.end = max([df3.loc['tasks', name].rate.max() for name in names])

    fig.xaxis.axis_label = 'cores'
    fig.yaxis.axis_label = 'tasks/s'
    fig.x_range.start = 0
    fig.y_range.start = 0

    fig.xaxis.ticker = x
    figs.append(fig)
    # panel = Panel(child=fig, title=axis_type)
    # panels.append(panel)
    
# tabs = Tabs(tabs=panels)
# show(tabs)
```

**Comment by mattpap** [comment]
> `sizing_mode='scale_width'` 

This is likely the problem, but I didn't confirm this.

**Comment by mrocklin** [comment]
Confirmed.  Thanks @mattpap

**Comment by mattpap** [comment]
OK, great. This particular sizing mode works best currently for standalone plots or simple layouts (like one row or column, but event then there are issues), and everything else is an untested territory. I will be fixing this either for 0.12.7 or 0.12.8, but first I need to do some more ground work.

**Comment by mattpap** [comment]
I will leave this open for my own bookkeeping, but it's likely that this is a duplicate of other issue(s).

**Comment by bryevdv** [comment]
I imagine this is resolved by #8085 but it's not possible to test without an MRE

**Comment by bryevdv** [comment]
Given that there will never be "unsatisfiable constraint" and several other tabs related issues have been confirmed fixed by #8085 I'm just going to close this as `norepro`
